R/01-mapPanelModule.R

```diff
@@ -80,10 +84,41 @@ mapPanelServer <- function(id) {
   moduleServer(
     id,
     function(input, output, session) {
+      image_list <- reactiveVal()
+      # load zip file
+      observe({
+        datapath <- input[["file"]]$datapath
+        if (substr(datapath, nchar(datapath) - 5 + 1, nchar(datapath)) != ".zipm") {
+          shinyjs::alert("Only .zipm files are supported!")
+        } else {
+          utils::unzip(datapath, exdir = tempdir())
+          image_list(convertJsonToDataFrame(file = paste0(tempdir(), "/image_list.json")))
+        }
+      }) %>% bindEvent(input[["file"]])
+
+      # fill group input
+      observe({
+        if (!is.null(image_list())) {
+          choices <- unique(image_list()$Group)
+        } else {
+          choices <- ""
+        }
+        updateSelectizeInput(
+          session = session,
+          inputId = "group_name-selectize",
+          choices = choices,
+          selected = ""
+        )
+      }) %>%
+        bindEvent(image_list(),
+          ignoreNULL = FALSE,
+          ignoreInit = TRUE
+        )
+
       # fill variable input
       observe({
         if (!is.null(input[["group_name-selectize"]])) {
-          choices <- image_list$Variable[image_list$Group_name == input[["group_name-selectize"]]]
+          choices <- image_list()$Variable[image_list()$Group == input[["group_name-selectize"]]]
         } else {
           choices <- ""
```

Comment:
As we discussed, I have changed it so that there are no more empty choices. This is because the json is checked for completeness when it is read in. If information is missing, a warning is displayed to the user and the choice is removed.